backend/substrapp/views/utils.py

```diff
@@ -217,3 +217,15 @@ def get_channel_name(request):
         return request.headers['Substra-Channel-Name']
 
     raise exceptions.BadRequestError('Could not determine channel name')
+
+
+def data_to_data_response(data):
```

Comment:
It looks like we want to call this before every response.
Wouldn't be more appropriate to use middleware in that particular case?

Advantages :
 - No need to modify all the views
 - Once the chaincode is adopting this change (replace pkhash with key) we only need to remove the middleware.

What do you think?

Reply:
We use this function for `create` asset call responses as we merge information from django and chaincode response.
I'm not familiar with middleware for this kind of stuff (overwrite one route response) and I don't know if it can be used for this kind of problem.
The pkhash concept is currently linked with use the django psql db. I don't know if it will be easy to migrate from it to something else :)
In addition, there are lot of similar code parts in multiple views in the backend that surely need to be refactor to something simpler.

Reply:
I have another view on this. 

I think the backend should own the return values of its interfaces. Instead, right now the backend proxies the chaincode response (and this PR adds a modification to this proxying).

I think a better solution would be to remove the `data_to_data_response` function entirely, and let each interface **explicitely** compute and return its return value. For instance:

```python
class TrainTupleViewSet(mixins.CreateModelMixin,
   [...]
   data = self._create(request) // from the chaincode
   response = { 'key': data['pkhash'] }
   return Response(response)
```

IMO this would be better because it's more explicit.

Reply:
@AurelienGasser Yep, it could be a better approach but looks like a bigger work than solving the issue of this PR ^^